### DISCUSSION TITLE
Launch chrome debugging

### DISCUSSION THREAD

**Discussion by jerpena** [coder/code-server#5035] [Q&A]
Has anyone here been able to get this to work? I always get a message saying "We can't launch a browser in debug mode from here. Open this workspace in VS Code on your desktop to enable debugging."

**Comment by lg-nineoneninethreenine** [comment]
There are a bunch of issues with debugging. I think this happens since code-server runs on the browser, so it can't access it since the client accessing is the browser. Possibly right, but I don't know.

**Reply by jsjoeio** [reply]
Yeah, not sure on this either. Does it work in Codespaces? There may be something we need to patch potentially to get it working.

**Comment by voscarmv** [comment]
Any updates on this? It would be awesome to be able to debug JS web apps from codespaces.

**Reply by code-asher** [reply]
No updates, I think this would need to be implemented upstream in https://github.com/microsoft/vscode

**Comment by coderzniv** [comment]
> Has anyone here been able to get this to work? I always get a message saying "We can't launch a browser in debug mode from here. Open this workspace in VS Code on your desktop to enable debugging."

same here

**Comment by bilogic** [comment]
I suspect this might be possible if we 

1. Manually launch Chrome in debug mode `chrome.exe --remote-debugging-port=9222 --user-data-dir=remote-debug-profile`
2. Manually port forward 9222 to the server running code-server and expose it as 9222
3. Run debugger with `Attach to browser`
```
   {
                "type": "chrome",
                "request": "attach",
                "name": "Attach to browser",
                "port": 9222
            },
```
